Mathlib/NumberTheory/Transcendental/Lindemann/Init/AnalyticalPart.lean

```diff
@@ -0,0 +1,206 @@
+  obtain ⟨c', hc', h'⟩ := P_le_aux p s c hc; clear c hc
+  let c₁ := max (Real.exp s.re) 1
+  let c₂ := max (Real.exp (Complex.abs s)) 1
+  let c₃ := max (Complex.abs s) 1
+  use c₁ * (c₂ * c' * c₃), by positivity
+  intro q hq
+  refine (h' q).trans ?_
+  simp_rw [mul_pow]
+  have le_max_one_pow {x : ℝ} : x ≤ max x 1 ^ q :=
+    (max_cases x 1).elim (fun h ↦ h.1.symm ▸ le_self_pow₀ h.2 hq)
+      fun h ↦ by rw [h.1, one_pow]; exact h.2.le
+  gcongr <;> exact le_max_one_pow
+
+/--
+Given a polynomial with integer coefficients `p` and a complex constant `s`, there is a nonnegative
+`c` such that for all nonzero `q ∈ ℕ`, `|P(X ^ (q - 1) * p ^ q, s)| ≤ c ^ q`.
+
+Note: Jacobson writes `h(x)` for `x ^ (q - 1) * p(x) ^ q` and `bⱼ` for its coefficients.
+-/
+private theorem exp_polynomial_approx_aux (p : ℤ[X]) (s : ℂ) :
+    ∃ c ≥ 0,
+      ∀ q ≠ 0, Complex.abs (P (map (algebraMap ℤ ℂ) (X ^ (q - 1) * p ^ q)) s) ≤ c ^ q := by
+  have : Bornology.IsBounded
+      ((fun x : ℝ ↦ max (x * abs s) 1 * Complex.abs (aeval (x * s) p)) '' Set.Ioc 0 1) := by
+    have h :
+      (fun x : ℝ ↦ max (x * abs s) 1 * Complex.abs (aeval (x * s) p)) '' Set.Ioc 0 1 ⊆
+        (fun x : ℝ ↦ max (x * abs s) 1 * Complex.abs (aeval (x * s) p)) '' Set.Icc 0 1 :=
+      Set.image_subset _ Set.Ioc_subset_Icc_self
+    refine (IsCompact.image isCompact_Icc ?_).isBounded.subset h
+    fun_prop
+  obtain ⟨c, h⟩ := this.exists_norm_le
+  simp_rw [Real.norm_eq_abs] at h
+  refine P_le _ s c (fun q x hx => ?_)
+  specialize h (max (x * abs s) 1 * Complex.abs (aeval (x * s) p)) (Set.mem_image_of_mem _ hx)
+  refine le_trans ?_ (pow_le_pow_left₀ (abs_nonneg _) h _)
+  simp_rw [Polynomial.map_mul, Polynomial.map_pow, map_X, eval_mul, eval_pow, eval_X, map_mul,
+    Complex.abs_pow, real_smul, map_mul, abs_ofReal, ← eval₂_eq_eval_map, ← aeval_def, abs_mul,
+    Complex.abs_abs, mul_pow, abs_of_pos hx.1]
+  refine mul_le_mul_of_nonneg_right ?_ (pow_nonneg (Complex.abs.nonneg _) _)
+  rw [← mul_pow, _root_.abs_of_nonneg (by positivity), max_def]
+  split_ifs with hx1
+  · rw [one_pow]
+    exact pow_le_one₀ (mul_nonneg hx.1.le (Complex.abs.nonneg _)) hx1
+  · push_neg at hx1
+    exact pow_le_pow_right₀ hx1.le (Nat.sub_le _ _)
+
+/--
+See equation (68), page 285 of [Jacobson, *Basic Algebra I, 4.12*][jacobson1974].
+
+Given a polynomial `p` with integer coefficients, we can find a constant `c : ℝ` and for each prime
+`q > |p₀|`, `n(q) : ℤ` and `g(q) : ℤ[X]` such that
+
+* `q` does not divide `n(q)`
+* `deg(g(q)) < q * deg(p)`
+* all complex roots `r` of `p` satisfy `|n(q) * e ^ r - q * g(q)(r)| ≤ c ^ q / (q - 1)!`
+
+In the proof of Lindemann-Weierstrass, we will take `p` to be a polynomial whose complex roots
+are the algebraic numbers whose exponentials we want to prove to be linearly independent.
+
+Note: Jacobson writes `f` for our `p`, `p` for our `q`, `N_p` for our `n(q)`, `g_p` for our `g(q)`
+and `M` for our `c` (modulo a constant factor).
```

Comment:
I wouldn't mind calling the polynomial `f` instead of `p`. Especially if there is also a prime floating around. Hopefully this shouldn't be hard, using the LSP rename feature.
And then the prime can be called `p` instead of `q`.
I don't care strongly about `M` vs `c`. (In fact, I would have a slight pref for `c`.)

If you agree, then please also rename the polynomials higher up in the file.